proxy/grpcproxy/lease.go

```diff
@@ -15,20 +15,71 @@
 package grpcproxy
 
 import (
+	"io"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"google.golang.org/grpc/metadata"
+
 	"golang.org/x/net/context"
+	"golang.org/x/time/rate"
 
 	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/etcdserver/api/v3rpc/rpctypes"
 	pb "github.com/coreos/etcd/etcdserver/etcdserverpb"
 )
 
 type leaseProxy struct {
-	client *clientv3.Client
+	client   *clientv3.Client
```

Comment:
using clientv3.Lease fails few tests that specifically use raw grpc `LeaseGrant()` with `pb.LeaseGrantRequest{ID: 1, TTL: 1})`. 
e.g
```go
lresp, err := toGRPC(clus.RandClient()).Lease.LeaseGrant(
		context.TODO(),
		&pb.LeaseGrantRequest{ID: 1, TTL: 1})
```
since the clientv3.Lease.Grant() doesn't not accept a lease ID, there is no way to satisfy a `LeaseGrant()` req for a specific lease ID.